File: 백준/Bronze/2798. 블랙잭/블랙잭.py
card, targetSum = map(int,input().split())
cardList = list(map(int,input().split()))

def blackJack(card, targetSum, cardList) :
    maxSum = 0
    for a in range(card-2) :
        for b in range(a+1,card-1) :
            for c in range(b+1, card) :
                curSum = cardList[a] + cardList[b] + cardList[c]
                if curSum == targetSum :
                    return targetSum
                if maxSum < curSum and curSum < targetSum :
                    maxSum = curSum
    return maxSum

print(blackJack(card, targetSum, cardList))

File: 백준/Bronze/2798. 블랙잭/블랙잭.js
const input = require('fs').readFileSync(process.platform === "linux" ? "/dev/stdin" : "./input.txt")
    .toString().trim().split('\n');

const [T, targetSum] = input[0].split(' ').map(Number);
const cardList = input[1].split(' ').map(Number);

function solution() {
    let maxSum = 0;
    function dfs(cardCnt, startIndex, sum) {
        if(cardCnt === 3) {
            if(sum <= targetSum && sum > maxSum) {
                maxSum = sum;
            }
            return;
        }
        for(let i = startIndex; i < cardList.length; i++) {
            dfs(cardCnt + 1, i + 1, sum + cardList[i]);
        }
    }
    dfs(0,0,0);
    return maxSum;
}

console.log(solution());